File: Decomposed_Data/Control/Cough/Classer.m
myDir = uigetdir; %gets directory
myFiles = dir(fullfile(myDir,'*.mat')); %gets all csv files in struct

FinalData = {};

for k = 1:length(myFiles)
  baseFileName = myFiles(k).name;
  fullFileName = string(baseFileName);

  fprintf(1, 'Now Reading -  %s\sn', fullFileName);
  initial = load(fullFileName);
  len = length(initial.decomposed_Data);
  for i=1:len
      data = initial.decomposed_Data{i,1};
      data = data';
      FinalData = [FinalData;data];
  end
end

save('CoughDataControl_ch1','FinalData');


FinalData = {};

for k = 1:length(myFiles)
  baseFileName = myFiles(k).name;
  fullFileName = string(baseFileName);

  fprintf(1, 'Now Reading -  %s\sn', fullFileName);
  initial = load(fullFileName);
  len = length(initial.decomposed_Data);
  for i=1:len
      data = initial.decomposed_Data{i,2};
      data = data';
      FinalData = [FinalData;data];
  end
end

save('CoughDataControl_ch2','FinalData');

FinalData = {};

for k = 1:length(myFiles)
  baseFileName = myFiles(k).name;
  fullFileName = string(baseFileName);

  fprintf(1, 'Now Reading -  %s\sn', fullFileName);
  initial = load(fullFileName);
  len = length(initial.decomposed_Data);
  for i=1:len
      data = initial.decomposed_Data{i,3};
      data = data';
      FinalData = [FinalData;data];
  end
end

save('CoughDataControl_ch3','FinalData');

File: Decomposed_Data/Control/Cough/Classer.py
import os
import numpy as np

# Get directory
my_dir = input("Enter directory path: ")  # You can replace this with a dialog box for directory selection

# Get all .mat files in the directory
my_files = [f for f in os.listdir(my_dir) if f.endswith('.mat')]

# Initialize FinalData
FinalData = []

# Process each file
for file_name in my_files:
    print("Now Reading - ", file_name)
    full_file_name = os.path.join(my_dir, file_name)
    initial = np.load(full_file_name, allow_pickle=True)
    for i in range(len(initial['decomposed_Data'])):
        data = initial['decomposed_Data'][i][0].T
        FinalData.append(data)

# Save FinalData for channel 1
np.save('CoughDataControl_ch1.npy', FinalData)

# Process each file again for channel 2
FinalData = []
for file_name in my_files:
    print("Now Reading - ", file_name)
    full_file_name = os.path.join(my_dir, file_name)
    initial = np.load(full_file_name, allow_pickle=True)
    for i in range(len(initial['decomposed_Data'])):
        data = initial['decomposed_Data'][i][1].T
        FinalData.append(data)

# Save FinalData for channel 2
np.save('CoughDataControl_ch2.npy', FinalData)

# Process each file again for channel 3
FinalData = []
for file_name in my_files:
    print("Now Reading - ", file_name)
    full_file_name = os.path.join(my_dir, file_name)
    initial = np.load(full_file_name, allow_pickle=True)
    for i in range(len(initial['decomposed_Data'])):
        data = initial['decomposed_Data'][i][2].T
        FinalData.append(data)

# Save FinalData for channel 3
np.save('CoughDataControl_ch3.npy', FinalData)
